### PR TITLE
5.3 make 5.1 and 5.2 work together

### PR DESCRIPTION
**Pull request by AdamRohdell**

### CODE REVIEW

**Comment by amandadehlen** [comment]
image
Skulle vara snyggare om den här sidan är centrerad.. men är egentligen inte din uppgift så bara om det är en quickfix

File: client/src/components/admin.js [hunk 1]
```diff
@@ -17,31 +21,206 @@ class Admin extends Component {
...
+    //function to create new items, is called when the button in adminNewMenuItem is pressed
+    makeNewItem() {
+        let temp = {
+            name: "",
+            id: this.getLatestMenuItemId(),
+            description: "",
+            availableToppings: [],
+            price: "",
+            status: "available",
+            amount: "",
```
Comment (steget) [inline client/src/components/admin.js:134]: amount is not used anymore. If it breaks some other part of the program I'd suggest you hardcode amount to that other part so this is working as intended. If not, then atleast a todo comment that says it should be removed
```suggestion
```

File: client/src/components/admin.js [hunk 2]
```diff
@@ -17,31 +21,206 @@ class Admin extends Component {
...
+            method: "POST",
+            headers: { "Content-Type": "application/json" },
+            body: JSON.stringify(item),
+        };
+
+        fetch("/api/add_menuItem", requestOptions);
+    }
+
+    //Update the current state to match the new updated menuItem, this is called from inside menuItemEditor
+    save(state) {
```
Comment (steget) [inline client/src/components/admin.js:111]: state is not a descriptive parameter, newItemInfo or something would fit better

File: client/src/components/admin.js [hunk 3]
```diff
@@ -17,31 +21,206 @@ class Admin extends Component {
...
+    //Triggered when the backbutton is pressed and also checks if it was a new item being created
+    //when canceling, if so then delete the new item as well.
+    backButtonPressed(id) {
+        if (this.state.newItemBeingCreated == true) {
+            this.deleteItem(id);
+        } else {
+            this.selectItem();
+        }
+        this.setState({ newItemBeingCreated: false });
+        console.log(this.state);
```
Comment (steget) [inline client/src/components/admin.js:172]:
```suggestion
```

File: client/src/components/menuItemEditor.js [hunk 4]
```diff
@@ -0,0 +1,203 @@
...
+                                onChange={this.priceChanged}
+                            ></input>
+                            <label className="ml-2">kr</label>
+                        </div>
+
+                        <div>
+                            <button
+                                type="button"
+                                className="btn btn-danger mt-3"
+                                syle={{ fontSize: "22px" }}
```
Comment (steget) [inline client/src/components/menuItemEditor.js:167]:
```suggestion
                                style={{ fontSize: "22px" }}
```

File: server.js [hunk 5]
```diff
@@ -38,41 +63,39 @@ app.get("/api/orders", (req, res) => {
...
+    res.end();
+});
+
+//Handle post request for updating menuItems from the admin page
+app.post("/api/update_menuItem", (req, res) => {
+    let menuItem = req.body;
+    let temp = menuItemArray.find((i) => i.id === menuItem.id);
+    let index = menuItemArray.indexOf(temp);
+    menuItemArray[index] = menuItem;
+    console.log(menuItemArray);
```
Comment (steget) [inline server.js:81]:
```suggestion
```

File: client/src/components/admin.js (hunk 1 above)
Comment (AdamRohdell) [reply client/src/components/admin.js:134]: Removed it, nothing broke so far.

**Pull request merged**